src/factorizations/blocklanczos.jl

```diff
@@ -0,0 +1,326 @@
+# BlockLanczos
+"""
+    struct BlockVec{T,S<:Number}
+
+Structure for storing vectors in a block format. The type parameter `T` represents the type of vector elements,
+while `S` represents the type of inner products between vectors.
+"""
+struct BlockVec{T,S<:Number}
+    vec::Vector{T}
+    function BlockVec{S}(vec::Vector{T}) where {T,S<:Number}
+        return new{T,S}(vec)
+    end
+end
+Base.length(b::BlockVec) = length(b.vec)
+Base.getindex(b::BlockVec, i::Int) = b.vec[i]
+function Base.getindex(b::BlockVec{T,S}, idxs::AbstractVector{Int}) where {T,S}
+    return BlockVec{S}([b.vec[i] for i in idxs])
+end
+Base.setindex!(b::BlockVec{T}, v::T, i::Int) where {T} = (b.vec[i] = v)
+function Base.setindex!(b₁::BlockVec{T}, b₂::BlockVec{T},
+                        idxs::AbstractVector{Int}) where {T}
+    return (b₁.vec[idxs] = b₂.vec;
+            b₁)
+end
+LinearAlgebra.norm(b::BlockVec) = norm(b.vec)
+function apply(f, block::BlockVec{T,S}) where {T,S}
+    return BlockVec{S}([apply(f, block[i]) for i in 1:length(block)])
+end
+function initialize(x₀, size::Int)
```

Comment:
This is the function I currently most dislike. `similar` and `randn!` are not part of the interface, and `initialize` of the Krylov iterator structs has little to do with this `initialize` method.

I would think that it's a user's responsibility to provide several starting vectors, so that you would actually never have to generate new random vectors from a single starting vector.